legal-api/src/legal_api/models/business.py

```diff
@@ -694,65 +695,68 @@ def get_alternate_names(self) -> dict:
         """
         alternate_names = []
 
-        # Get start date value for aliases from filing.effective_date that's tied to
-        # the transaction that created the alias
+        # Fetch aliases and related filings in a single query
         alias_version = version_class(Alias)
-        aliases = db.session.query(alias_version). \
-            filter(alias_version.id.in_([a.id for a in self.aliases]),
-                   alias_version.end_transaction_id == None  # noqa: E711; pylint: disable=singleton-comparison
-                   ).all()
-        for alias in aliases:
-            filing = db.session.query(Filing). \
-                filter(Filing.transaction_id == alias.transaction_id
-                       ).one_or_none()
+        filing_alias = aliased(Filing)
```

Comment:
This function was changed using a combination of my brain and AI's brain 😆 

The idea is, we're kind of doing a single query now and using joins. So, it's decreasing the amount of roundtrips.